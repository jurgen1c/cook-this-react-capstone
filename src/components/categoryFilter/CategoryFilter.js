import './category.css'
const CategoryFilter = ({handleChange, categories}) => {
  return (
      <select name="categories" onChange={handleChange}>
        <option selected disabled>Select a category</option>
        {categories.map(item => <option key={`${item.strCategory}_option`} value={item.strCategory}>{item.strCategory}</option>)}
      </select>
  )
}


CategoryFilter.propTypes = {
  categories: PropTypes.array.isRequired,
  handleChange: PropTypes.func.isRequired,
};
export default CategoryFilter
