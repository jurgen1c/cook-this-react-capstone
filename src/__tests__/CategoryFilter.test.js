import React from 'react';
import renderer from 'react-test-renderer';
import CategoryFilter from '../components/categoryFilter/CategoryFilter';

const categories = [
  {
    strCategory: 'Category'
  },
  {
    strCategory: 'Category1'
  }
]

const handleChange = () => {}
test('Componenet renders correctly', () => {
  const component = renderer.create(
    <CategoryFilter categories={categories} handleChange={handleChange}/>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});