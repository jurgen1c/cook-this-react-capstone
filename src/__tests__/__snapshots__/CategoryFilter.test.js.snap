// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componenet renders correctly 1`] = `
<select
  name="categories"
  onChange={[Function]}
>
  <option
    disabled={true}
    selected={true}
  >
    Select a category
  </option>
  <option
    value="Category"
  >
    Category
  </option>
  <option
    value="Category1"
  >
    Category1
  </option>
</select>
`;
